Issue/24/interact with mysql database

jbkalmbach opened the pull request:
As outlined in #24. Before merging, discuss with @jchiang87 about user input for database connection details.

jbkalmbach committed:
Added basic database interface.

jbkalmbach committed:
Added basic database interface.

jbkalmbach committed:
Added object lookup by ra, dec.

jbkalmbach committed:
Added object lookup by ra, dec.

jbkalmbach committed:
Updated example to include database interface. Adjust monitor to work with local tables still.

jbkalmbach committed:
Updated example to include database interface. Adjust monitor to work with local tables still.

jbkalmbach committed:
Merge branch 'issue/24/interact-with-mysql-database' of https://github.com/DarkEnergyScienceCollaboration/Monitor into issue/24/interact-with-mysql-database

jchiang87 commented on python/desc/monitor/dbConnection.py:
```diff
@@ -0,0 +1,64 @@
+
+class dbInterface(object):
+
+    def __init__(self):
```
@jbkalmbach Regarding giving users the ability to select the database, host, and port, I guess the minimal thing to do would be to make these arguments to the `dbInterface` constructor here and also in the `LightCurve.build_lightcurve_from_db` method below.  To be explicit, the code here would be

```
    def __init__(self, database, host, port, driver):
        self._dbo = DBObject(database=database, host=host, port=port, driver=driver)
```

rbiswas4 replied:
@jchiang87 we have been discussing this too! We would also like 
- to expose those parameters to the user 
- and thinking ahead to cases where the databases will not be read-only, avoid putting credential information in the source code.  @jbkalmbach has example snippets (based on other lsst codes) to do this
  and obviously a PR would be a good place to discuss this more broadly within the group, and get further ideas.

jchiang87 replied:
I thought that the `.lsst/db-auth.paf` mechanism, which we're using here, was sufficient, since that provides the user and password in a secure way.  Are you also worried about the host and port info?

rbiswas4 replied:
That is in fact the mechanism we were using planning on explicitly in the code with default arguments to those parameters. I did not notice that this was already taken care of. I don't worry much about the hostname, though this can be similarly removed.

jchiang87 commented on python/desc/monitor/monitor.py:
```diff
@@ -110,6 +116,54 @@ def build_lightcurve(self, objid):
                     lightcurve['zpsys'].append('ab')
         self.lightcurve = Table(data=lightcurve)
 
+    def build_lightcurve_from_db(self, objid=None, ra_dec=None, tol=0.005):
```
Here the change would be

```
    def build_lightcuve_from_db(self, objid=None, ra_dec=None, tol=0.005, database='DESC_Twinkles_Level_2',
                                                  host='scidb1.nersc.gov', port=3306, driver='mysql'):
```

and the last four would be passed to the `dbi(...)` call below.   It's a little inconvenient to provide the db details with every call to `.build_lightcurve_from_db`, which is one of the reasons I decided to use the Factory pattern in my implementation of `LightCurveFactory` in Twinkles.   We might consider doing something like that here.  Also, the Factory pattern makes it more natural, I think, to serve up immutable `LightCurve` objects so that the light curve data can be provided through the constructor or a private method.  There may be some db connection management issues associated with creating the connection objects locally like you're doing here.  Those concerns could be mitigated again with the Factory pattern.

In the near term, it's probably fine to make the minimal changes so that we can merge this into master and people can start using it.

jbkalmbach committed:
Added user input for database parameters

jbkalmbach committed:
Updated README and example

jbkalmbach commented:
Ok, I think I've made the necessary changes for now and asked Rahul to test and review the code. As soon as he signs off we'll merge.

jbkalmbach committed:
Added future absolute_import

jbkalmbach committed:
Updated import of dbConnection to py3 format.

rbiswas4 committed:
renamed:    __init__.py -> init.py.swp
	modified:   monitor/__init__.py
	modified:   monitor/dbConnection.py
	modified:   ../../tests/test_Monitor.py

rbiswas4 committed:
Trying to fix travis failures
	renamed:    python/desc/init.py.swp -> python/desc/__init__.py
	modified:   python/desc/monitor/__init__.py

rbiswas4 committed:
Fixed pep8 changes
	modified:   examples/Light_curve_visualization/example.py
